### PR TITLE
Post title: handle paste as blocks

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

This PR handles pasting in the post title (post editor) a bit better. Before everything would be pasted inline as the title, now we parse the pasted content as blocks and paste the block after the title. If the first block is a heading or a paragraph AND the title is empty, the content of this first block will be used for the title. I've left out more complex splitting, for example when there would be already text in the title and the selection is in the middle. Ideally we should be using RichText or PlainText version 2 for the post title, which has built in splitting and paste handling, but in order to use this, the post title should be a real block... which I think we plan to do at some point? I recall having run into problems with the post title block a few times now because it's not a real block, especially with writing flow. Cc @youknowriad @mtias.

To do: maybe add some e2e tests.

## How has this been tested?

Paste a few blocks of text from a page into the post title. Also try to paste a small string of text, try with title already filled in etc.

## Types of changes

Enhancement

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/editor/src/components/post-title/index.js [hunk 1]
```diff
@@ -60,6 +62,57 @@ class PostTitle extends Component {
...
+				( firstBlock.name === 'core/heading' ||
+					firstBlock.name === 'core/paragraph' )
+			) {
+				onUpdate( firstBlock.attributes.content );
+				onInsertBlockAfter( content.slice( 1 ) );
+			} else {
+				onInsertBlockAfter( content );
+			}
+		}
+	}
```
Comment (youknowriad) [inline packages/editor/src/components/post-title/index.js:114]: Do you think there's logic to be shared between this and the other place we're using it today (RichText?)?

**Review by mapk — APPROVED** [review]
I just tested this out. It works wonderfully from a functional perspective! ❤️ 

pasting

File: packages/editor/src/components/post-title/index.js (hunk 1 above)
Comment (mtias) [reply packages/editor/src/components/post-title/index.js:114]: I think as soon as we turn the title into a block a few things would have to be consolidated.

Comment (ellatrix) [reply packages/editor/src/components/post-title/index.js:114]: Right, there is logic that can be shared, but we should instead aim to turn the post title into a block, which means the logic will automatically be shared through either RichText or PlainText v2.

**Comment by ellatrix** [comment]
Let's merge this and iterate. I was looking at e2e tests quick, but we need to emulate all clipboard behaviour on textarea and I didn't get it to work immediately.

**Comment by mtias** [comment]
Thanks @ellatrix ! Love these small quality-of-life improvements :)

**Pull request merged**